repoman/git/repository.py

```diff
@@ -358,11 +370,9 @@ def push(self, orig, dest, rev=None, ref_name=None, force=False):
             refspec = "%s:%s" % (rev, ref_name)
 
         if all_tags_option:
-            self._git("push", dest, refspec, all_tags_option,
-                      all_notes_refspec, f=force)
+            self._git("push", dest, refspec, all_tags_option, f=force)
         else:
-            self._git("push", dest, refspec, all_notes_refspec,
-                      f=force)
+            self._git("push", dest, refspec, f=force)
         return self.tip()
```

Comment:
The changes in this push method are the main ones of this PR.

Reply:
Cool. Just a style question, as before there is no explicit `push_notes()` as they were pushed by default. Now they are not, but we still can make a generic `push()` using the git reference for all notes. But that is not very explicit. What do you think about a `push_notes()` method?

Reply:
I would just let the user use `push()` method with the explicit `ref_name` parameter he/she wants to use for pushing. In the case of notes, it could be like `ref_name="refs/notes/*"`

Taking a look at the tests we already had, there is a specific test that covers that use case using `ref_name`.

Moreover, keeping the push() method it is kept the interface-like Repository class it is defined in `repoman/repository.py`  https://github.com/tuenti/python-repoman/blob/master/repoman/repository.py#L351